piv/src/main/java/com/yubico/yubikit/piv/PivSession.java

```diff
@@ -433,13 +466,38 @@ public byte[] calculateSecret(Slot slot, ECPublicKey peerPublicKey) throws IOExc
      * @throws ApduException        in case of an error response from the YubiKey
      * @throws BadResponseException in case of incorrect YubiKey response
      */
+    @Deprecated
     public byte[] calculateSecret(Slot slot, ECPoint peerPublicKey) throws IOException, ApduException, BadResponseException {
         KeyType keyType = peerPublicKey.getAffineX().bitLength() > 256 ? KeyType.ECCP384 : KeyType.ECCP256;
         byte[] encodedPoint = new PublicKeyValues.Ec(((KeyType.EcKeyParams)keyType.params).getCurveParams(), peerPublicKey.getAffineX(), peerPublicKey.getAffineY()).getEncodedPoint();
         Logger.debug(logger, "Performing key agreement with key in slot {} of type {}", slot, keyType);
         return usePrivateKey(slot, keyType, encodedPoint, true);
     }
 
+    /**
+     * Perform an ECDH operation with a given public key to compute a shared secret.
+     *
+     * @param slot            the slot containing the private EC key
+     * @param publicKeyValues the peer public key for the operation
+     * @return the shared secret, comprising the x-coordinate of the ECDH result point.
+     * @throws IOException              in case of connection error
+     * @throws ApduException            in case of an error response from the YubiKey
+     * @throws BadResponseException     in case of incorrect YubiKey response
+     * @throws NoSuchAlgorithmException in case of unsupported PublicKey type
+     */
+    public byte[] calculateSecret(Slot slot, PublicKeyValues publicKeyValues) throws IOException, ApduException, BadResponseException, NoSuchAlgorithmException, InvalidKeySpecException {
+        PublicKey publicKey = publicKeyValues.toPublicKey();
```

Comment:
We need to be able to do this without calling toPublicKey(), as that will require a JCA provider that supports the algorithm, and that shouldn't be needed.
Instead, we should be able to get the encoded point directly from the PublicKeyValues object.